src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -35,15 +35,47 @@ final class Dotenv
     private $data;
     private $end;
     private $values;
-    private $usePutenv;
+    private $envKey;
+    private $debugKey;
+    private $prodEnvs = ['prod'];
+    private $usePutenv = false;
 
     /**
-     * @var bool If `putenv()` should be used to define environment variables or not.
-     *           Beware that `putenv()` is not thread safe, that's why this setting defaults to false
+     * @param string $envKey
      */
-    public function __construct(bool $usePutenv = false)
+    public function __construct($envKey = 'APP_ENV', string $debugKey = 'APP_DEBUG')
+    {
+        if (\in_array($envKey = (string) $envKey, ['1', ''], true)) {
```

Comment:
`is_bool($envKey)` ?

Reply:
wouldn't make a diff, and this style makes the block "removable" as is in 6.